app/jobs/shipit/update_github_last_deployed_ref_job.rb

```diff
@@ -0,0 +1,39 @@
+module Shipit
+  class UpdateGithubLastDeployedRefJob < BackgroundJob
+    queue_as :default
+
+    DEPLOY_PREFIX = 'shipit-deploy'.freeze
+
+    def perform(stack)
+      stack_sha = stack.deploys.where(status: "success").last&.until_commit&.sha
```

Comment:
Are we expecting any `nil`s here? If so we should probably bail out of the job if `stack_sha` is nil

```ruby
return unless stack_sha
```

Reply:
I think a nil is possible in a somewhat rare edge case: A new stack, where the first deploy was successful (and thus enqueues the job) but is immediately reverted.

In this situation the Rollback deploy could call `Rollback.update_release_status`, which would mark the only successful deploy as unsuccessful, and the Rollback deploy can still be pending/running/validating when the Job finally executes.

The stack technically doesn't need to be new, but merely needs all prior deploys to the above to not be 'success'. Not that it makes it any more likely.

This is fairly contrived, but not _too_ crazy, so I think it's OK to guard here. But if you think I'm being a bit too paranoid (or misread something) let me know. I've simply had bad luck with deploying faulty changes over the past couple of weeks 😉.

Reply:
```ruby
if stack_sha.can_be_nil?
  add('return unless stack_sha')
else
  remove('&').from_method_calls
end
```

Since it can be `nil`, sounds like we should do the first option 😉 
After that this should be good to 🚢